### DISCUSSION TITLE
Got many suggestions...

### DISCUSSION THREAD

**Discussion by A983Hub** [ikemen-engine/Ikemen-GO#877] [Feature Request] [answered]
### 1. Achievements System
The User can add achievements in some sort of .def file, with images, descriptions and everything. Can be unlocked by playing the game, beating characters' arcade routes, beating a story mode arc, etc...
### 2. Custom Music System
This is kind of similar to the music system in Super Smash Bros. **Pick a stage, then press a certain button. A list of soundtracks (Specified by the user in a .def file) will show up, and it can show options to pick the music, play it, and stop or pause it.**
### 3. Tournament Mode
**There can be like tournaments set up by the player and you can watch the characters fight in the tournament until there is one character standing.** You can set up how the tournament is, Like how many players are there.
### 4. Multiple Storyboards
Let's say you want to make space for a fake loading screen in your IKEMEN build, but it only supports the logo and intro, and you have already put those in. Kinda sucks, y'know? How about this... **you can add multiple storyboards to the intro/logo section. And probably some other places, idk.**
### 5. Stats/User Profile Card
There can be like some sort of stats mode where you can view the stats you have made in your save file (Battles played, Story Mode Arcs beaten, Battles Won, Battle Lost, etc...), and you can put a nickname, card background and even an avatar. Can be from an .def file with folder, or through the fighters themselves.
### 6. Dynamic Map Stage Select Screen
**This is optional.** There could be like, a map, and you can put an image over that said map, and you can set which stages are placed on the map. _(P.S. there can be info for the stage, like weather.)_ And when you pick that part of the map, BOOM! VS. Screen, Battle Begins.
### 7. Dynamic Victory Screen
**Like the Map Stage Select Screen, this is optional.** The winning character does their victory pose and somewhere in the Victory Screen, the other characters will do their lose poses. Again, sort of like Smash Bros.
### 8. Shop
There can be like a shop to buy stuff like Menu Themes, Stages, Characters and Menu Backgrounds. **How can you get currency for the shop? Well, by beating Arcade Routes, Story Mode Arcs or Beating Battles!**
### 9. Challenge Mode
**There can be like challenges (Again, set by a .def file) in which you have to set a certain criteria.** (For example: Beat this character, beat this boss, beat these characters but with draining health, etc)
### 10. Stage Hazards
This could be kind of annoying, but Stage Hazards help make the stage feel... **dynamic and unique. Let's say, you're on a team battle, and a stage hazard (Like some sort of fireball or dragon) attacks a certain character.**

I'm probably sure 50% or less sure if these suggestions might not make it into IKEMEN. If it's not going to be included, then, that's ok, these are just my own suggestions to make the game feel more like a full game of sorts.

**Comment by Windblade-GR01** [comment, marked as answer]
Achievements System exists in some form with `stats.json`
Dynamic Map Stage Select Screen is posible using the stage portrait feature.
Dynamic Victory Screen are posible using animated portrait.
Challenge Mode is currently in development by two4teezee.
Stage Hazards are posible via the interactive stage feature.

**Comment by A983Hub** [comment]
> Achievements System exists in some form with `stats.json`
> Dynamic Map Stage Select Screen is posible using the stage portrait feature.
> Dynamic Victory Screen are posible using animated portrait.
> Challenge Mode is currently in development by two4teezee.
> Stage Hazards are posible via the interactive stage feature.

Alright then!
